docs/zkapps/o1js/keccak.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: Keccak
+hide_title: true
+description: A comprehensive guide on how to use Keccak-based hashes in o1js.
+keywords:
+  - keccak
+  - sha3
+  - hash
+  - ethereum
+  - mina blockchain
+  - blockchain technology
+  - data structures
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Keccak (SHA-3)
+
+Keccak is a flexible cryptographic hash function that provides more security than traditional SHA hash algorithms.
+
+## What is Keccak?
+
+Developed by a team of cryptographers from Belgium, Keccak was the winning submission for the National Institute of Standards and Technology (NIST) SHA-3 competition. Keccak was standardized as SHA-3. Today, Keccak is used in many different applications, including Ethereum and many others.
+
+Specifically, Keccak is used in the Ethereum ecosystem to hash addresses, transactions, and blocks. It is also used to hash the state trie, which is a data structure that stores the state of the Ethereum blockchain.
+Because of the common usage of Keccak in Ethereum, it is a key component of o1js that you can use to verify Ethereum transactions and blocks in o1js.
+
+## Keccak and Poseidon
+
+As an o1js developer, you are likely familar with the [Poseidon](https://o1-labs.github.io/proof-systems/specs/poseidon.html) zero knowledge native hash function. Poseidon operates over the native [Pallas base field](https://electriccoin.co/blog/the-pasta-curves-for-halo-2-and-beyond/) and uses parameters generated specifically for Mina which makes Poseidon the most efficient hash function available in o1js.
+
+In contrast, Keccak is a hash function that requires binary arithmetic. It operates over binary data and is not native to most zero knowledge proofs. For this reason, Keccak is not as efficient as Poseidon, but it is still very useful for verifying Ethereum transactions and blocks.
+So, when you choose what hash function to use, it is important to consider the use case and the data that needs to be hashed.
+
+## Basic usage
+
+Keccak is available in the following configurations that are available under the `Hash` namespace in o1js:
+
+- `Hash.SHA3_256`: NIST SHA3 hash function with output size of 256 bits.
+- `Hash.SHA3_385`: NIST SHA3 hash function with output size of 384 bits.
+- `Hash.SHA3_512`: NIST SHA3 hash function with output size of 512 bits.
+- `Hash.Keccak256`: pre-NIST Keccak hash function with output size of 256 bits. _This configuration used to hash blocks, transactions, and more in Ethereum._
+- `Hash.Keccak384`: pre-NIST Keccak hash function with output size of 384 bits.
+- `Hash.Keccak512`: pre-NIST Keccak hash function with output size of 512 bits.
+
+Because Keccak operates over binary data instead of native Field elements like Poseidon, o1js uses the `Bytes` type. `Bytes` is a fixed-length array of bytes that can be used to represent binary data.
```

Comment:
@Trivo25  I like your original writing for the blog post "new type" but for docs, let's write timeless content 
our docs content style:
> Write [timeless documentation](https://developers.google.com/style/timeless-documentation) that avoids words and phrases that anchor the documentation to a point in time or assume knowledge of prior or future products and features. Words to avoid: now, new, soon, not yet, latest, future, and currently.